lib/project/nativescript-project.ts

```diff
@@ -0,0 +1,127 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import path = require("path");
+import util = require("util");
+
+import frameworkProjectBaseLib = require("./framework-project-base");
+import MobileHelper = require("../common/mobile/mobile-helper");
+
+export class NativeScriptProject extends frameworkProjectBaseLib.FrameworkProjectBase implements Project.IFrameworkProject {
+	constructor(private $config: IConfiguration,
+		$fs: IFileSystem,
+		$logger: ILogger,
+		private $projectConstants: Project.IProjectConstants,
+		private $projectFilesManager: Project.IProjectFilesManager,
+		$resources: IResourceLoader,
+		private $templatesService: ITemplatesService) {
+		super($logger, $fs, $resources);
+	}
+
+	public get name(): string {
+		return this.$projectConstants.TARGET_FRAMEWORK_IDENTIFIERS.NativeScript;
+	}
+
+	public get capabilities(): IProjectCapabilities {
+		return {
+			build: true,
+			buildCompanion: true,
+			deploy: true,
+			simulate: false,
+			livesync: false,
+			livesyncCompanion: true,
+			updateKendo: false
+		};
+	}
+
+	public get defaultProjectTemplate(): string {
+		return this.$config.DEFAULT_NATIVESCRIPT_PROJECT_TEMPLATE;
+	}
+
+	public get liveSyncUrl(): string {
+		return "nativescript://";
+	}
+
+	public get requiredAndroidApiLevel(): number {
+		return 17; // 4.2 JellyBean
+	}
+
+	public get configFiles():  Project.IConfigurationFile[] {
+		var allConfigFiles = this.$projectFilesManager.availableConfigFiles;
+		return [
+			allConfigFiles["ios-info"],
+			allConfigFiles["android-manifest"]
+		]
+	}
+
+	public getTemplateFilename(name: string): string {
+		return util.format("Telerik.Mobile.NativeScript.%s.zip", name);
+	}
+
+	public alterPropertiesForNewProject(properties: any, projectName: string): void {
+		this.alterPropertiesForNewProjectBase(properties, projectName);
+	}
+
+	public projectTemplatesString(): IFuture<string> {
+		return this.$templatesService.getTemplatesString(/.*Telerik\.Mobile\.NativeScript\.(.+)\.zip/);
+	}
+
+	public getProjectFileSchema(): IFuture<any> {
+		return this.getProjectFileSchemaByName(this.name);
+	}
+
+	public getFullProjectFileSchema(): IFuture<any> {
+		return this.getFullProjectFileSchemaByName(this.name);
+	}
+
+	public getProjectTargets(projectDir: string): IFuture<string[]> {
+		var dir = path.join(projectDir, "app");
+		var fileMask = /^bootstrap\.(\w*)\.js$/i;
+
+		return this.getProjectTargetsBase(dir, fileMask);
+	}
+
+	public adjustBuildProperties(buildProperties: any, projectData?: IProjectData): any {
+		return buildProperties;
+	}
+
+	public ensureAllPlatformAssets(projectDir: string, frameworkVersion: string): IFuture<void> {
+		return (() => {
+
+		}).future<void>()();
+	}
+
+	public getSimulatorParams(projectDir: string, projectData: IProjectData, simulatorPackageName: string): IFuture<string[]> {
+		return null;
```

Comment:
Why not returning [] (as the other implementation does)? This will allow linq-style code without warring too much about nulls.